Remotion/Web/Development.WebTesting/WebDriver/Factories/Edge/EdgeBrowserFactory.cs

```diff
@@ -0,0 +1,94 @@
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.IO;
+using Coypu.Drivers;
+using JetBrains.Annotations;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Remote;
+using Remotion.Utilities;
+using Remotion.Web.Development.WebTesting.BrowserSession;
+using Remotion.Web.Development.WebTesting.BrowserSession.Edge;
+using Remotion.Web.Development.WebTesting.WebDriver.Configuration;
+using Remotion.Web.Development.WebTesting.WebDriver.Configuration.Chromium;
+
+namespace Remotion.Web.Development.WebTesting.WebDriver.Factories.Edge
+{
+  /// <summary>
+  /// Responsible for creating a new Edge browser, configured based on <see cref="IChromiumConfiguration"/>.
+  /// </summary>
+  public class EdgeBrowserFactory : BrowserFactoryBase
+  {
+    private const string c_edgePoliciesPath = @"SOFTWARE\Policies\Microsoft\Edge";
+
+    private readonly IChromiumConfiguration _edgeConfiguration;
+    private readonly ExtendedChromiumOptions _extendedEdgeOptions;
+    private readonly IBrowserSessionCleanUpStrategy _registryCleanUpStrategy;
+
+    public EdgeBrowserFactory ([NotNull] IChromiumConfiguration edgeConfiguration)
+    {
+      ArgumentUtility.CheckNotNull ("edgeConfiguration", edgeConfiguration);
+
+      _edgeConfiguration = edgeConfiguration;
+      _extendedEdgeOptions = _edgeConfiguration.CreateChromiumOptions();
+
+      var registryCleanUpStrategyFactory = new ChromiumSecurityWarningsRegistryCleanUpStrategyFactory (_edgeConfiguration.DisableSecurityWarningsBehavior, c_edgePoliciesPath);
+      _registryCleanUpStrategy = registryCleanUpStrategyFactory.CreateCleanUpStrategy();
+    }
+
+    protected override IBrowserSession CreateBrowserSession (Coypu.BrowserSession session, int driverProcessID)
+    {
+      ArgumentUtility.CheckNotNull ("session", session);
+
+      return new EdgeBrowserSession (
+          session,
+          _edgeConfiguration,
+          driverProcessID,
+          new[]
+          {
+              _registryCleanUpStrategy,
+              new ChromiumUserDirectoryCleanUpStrategy (_edgeConfiguration, _extendedEdgeOptions.UserDirectory)
+          });
+    }
+
+    protected override RemoteWebDriver CreateWebDriver (DriverService driverService)
+    {
+      ArgumentUtility.CheckNotNull ("driverService", driverService);
+
+      return new ChromeDriver ((ChromeDriverService) driverService, _extendedEdgeOptions);
+    }
+
+    protected override Browser GetBrowser ()
+    {
+      return Browser.Chrome;
+    }
+
+    protected override DriverService CreateDriverService ()
+    {
+      var driverDirectory = Path.GetDirectoryName (_edgeConfiguration.DriverBinaryPath);
+      var driverExecutable = Path.GetFileName (_edgeConfiguration.DriverBinaryPath);
+
+      var driverService = ChromeDriverService.CreateDefaultService (driverDirectory, driverExecutable);

```

Comment:
When do we get the EdgeDriverService?

Reply:
https://github.com/re-motion/Trunk-SVN2Git/pull/17#discussion_r335533959